lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -725,17 +784,169 @@ struct GrandCentralPass : public GrandCentralBase<GrandCentralPass> {
+                                               AnnoPathValue &companionTarget,
+                                               StringAttr viewName) {
+  if (isa<FExtModuleOp>(refSendTarget.ref.getOp())) {
+    refSendTarget.ref.getOp()->emitError(
+        "RefType source cannot be an extern module port, it must be moved to "
+        "the instance of the extern module");
+    return {};
+  }
+
+  auto refSendModule = cast<FModuleOp>(refSendTarget.ref.getModule());
+  // The value that must be the argument to the RefSendOp. This can either be a
+  // module port or the result of any namable operation.
+  Value refSendBase;
+  if (refSendTarget.ref.getImpl().isOp())
+    refSendBase = refSendTarget.ref.getImpl().getOp()->getResult(0);
+  // Note: This assumes that only the result 0 of the op must be connected.
+  // There is no mechanism currently to handle mulitple result operations.
+  else if (refSendTarget.ref.getImpl().isPort())
+    refSendBase =
+        refSendModule.getArgument(refSendTarget.ref.getImpl().getPortNo());
+  auto refSendBuilder = ImplicitLocOpBuilder::atBlockEnd(
+      refSendModule.getLoc(), refSendModule.getBodyBlock());
+  // If it is an aggregate, and only a subfield is required, then insert the
+  // appropriate number of Subfield and SubIndex to get the subfield value.
+  refSendBase =
+      getValueByFieldID(refSendBuilder, refSendBase, refSendTarget.fieldIdx);
+  // Create the refSend, a remote XMR to read `refSendBase`.
+  // Note: No DontTouch added, so constant-prop and cse are allowed.
+  auto sendVal = refSendBuilder.create<RefSendOp>(refSendBase);
+  FModuleOp lcaModule = parentModule;
+  SmallVector<InstanceOp> pathFromSrcToCompanion;
+  // Compute the path and set `pathFromSrcToCompanion`.
+  if (computeRefPortsPathViaLCA(refSendTarget, companionTarget,
+                                pathFromSrcToCompanion, lcaModule, state)
+          .failed())
+    return {};
+  // Now drill ports to connect the `sendVal` to the `wireTarget`.
+  auto remoteXMR = borePortsOnPath(
+      pathFromSrcToCompanion, lcaModule, sendVal, viewName.getValue(),
+      state.instancePathcache,
+      [&](FModuleLike mod) -> ModuleNamespace & {
+        return state.getNamespace(mod);
+      },
+      &state.targetCaches);
+  // Create the RefResolveOp in the companion.
+  FModuleOp companionModule = cast<FModuleOp>(companionTarget.ref.getOp());
+  auto compBuilder = ImplicitLocOpBuilder::atBlockEnd(
+      companionModule.getLoc(), companionModule.getBodyBlock());
+  auto refResolve = compBuilder.create<RefResolveOp>(remoteXMR);
+  auto newName = [&](FModuleOp nameForMod) {
+    return StringAttr::get(
+        state.circuit.getContext(),
+        state.getNamespace(nameForMod)
+            .newName("view_" + viewName.getValue() + "refPort"));
+  };
+
+  auto node = compBuilder.create<NodeOp>(refResolve.getResult().getType(),
+                                         refResolve, newName(companionModule));
+  node.dropName();
```

Comment:
Since the name is what's returned for use identifying the result of the resolve (and the code calling this uses it in the verbatim), I don't think the name should be droppable?

Reply:
The node name is generated by this pass, so I don't think we need to preserve it, and it can also be optimized away. We do add a `DontTouch` to it to ensure it is preserved till the `GrandCentral` pass. The `DontTouch` is then dropped by that pass.